src/controller/CHIPDeviceControllerFactory.cpp

```diff
@@ -438,6 +440,13 @@ void DeviceControllerSystemState::Shutdown()
         mFabricTableDelegate = nullptr;
     }
 
+    if (mBDXTransferServer != nullptr)
+    {
+        mBDXTransferServer->Shutdown();
+        chip::Platform::Delete(mBDXTransferServer);
```

Comment:
We cannot delete the BDX transfer server from here, since that assumes it is OK to delete here, and that it was allocated with chip::Platform::New. This should be injectable by a caller who knows the lifetime.

Reply:
I am a little bit confused. The lifetime is dictated by the device controller factory being initialized/shutdown no ? (ignoring https://github.com/project-chip/connectedhomeip/issues/16231 as this is a more global issue).

Reply:
do we have API guarantees that mBDXTransferServer was allocated with PlatformNew() ?

Reply:
At the moment this is only initialised internally in `DeviceControllerFactory.cpp` . We could change that if needed but that it how it works for now.